central/signatureintegration/datastore/singleton.go

```diff
@@ -1,21 +1,48 @@
 package datastore
 
 import (
+	"context"
+
+	"github.com/pkg/errors"
 	"github.com/stackrox/rox/central/globaldb"
 	policyDataStore "github.com/stackrox/rox/central/policy/datastore"
+	"github.com/stackrox/rox/central/signatureintegration/store"
 	pgStore "github.com/stackrox/rox/central/signatureintegration/store/postgres"
+	"github.com/stackrox/rox/pkg/features"
+	"github.com/stackrox/rox/pkg/sac"
 	"github.com/stackrox/rox/pkg/sync"
+	"github.com/stackrox/rox/pkg/utils"
 )
 
 var (
 	once     sync.Once
 	instance DataStore
 )
 
+func createDefaultRedHatSignatureIntegration(siStore store.SignatureIntegrationStore) {
+	ctx := sac.WithGlobalAccessScopeChecker(context.Background(), sac.AllowAllAccessScopeChecker())
+	if _, exists, err := siStore.Get(ctx, store.DefaultRedHatSignatureIntegration.GetId()); err != nil {
+		utils.Should(errors.Wrap(err, "unable to detect if default Red Hat signature integration exists"))
+		return
+	} else if exists {
+		// Nothing to do, integration exists.
+		return
+	}
```

Comment:
How about this
```suggestion
	if exists, err := siStore.Exists(ctx, store.DefaultRedHatSignatureIntegration.GetId()); exists {
		utils.Should(errors.Wrap(err, "unable to detect if default Red Hat signature integration exists"))
		return
	}
```

Reply:
I'm not sure I understand this. The idea is to warn if we couldn't check if it exists, and just return if it exists. Do I understand correctly that your suggestion would warn if the integration exists?

I replicated the logic on the image integration side: 

https://github.com/stackrox/stackrox/blob/858cda1/central/imageintegration/datastore/singleton.go#L133-L145

Reply:
After this I both simplified the logic to always upsert and removed the feature flag logic. Resolving as this is now outdated.